<?xml version="1.0"?>
<launch>
    <!-- 参数配置文件路径 -->
    <arg name="config_file" default="$(find lidar_slam_from_scratch)/config/params.yaml"/>
    
    <!-- LiDAR SLAM 节点 -->
    <node pkg="lidar_slam_from_scratch" type="lidar_slam_node" name="lidar_slam_node" output="screen">
        <rosparam command="load" file="$(arg config_file)"/>
    </node>
    
    <!-- RViz 可视化 -->
    <arg name="rviz" default="true"/>
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rviz" 
          args="-d $(find lidar_slam_from_scratch)/config/slam.rviz"/>
</launch>
